src/ordering.f90

```diff
@@ -0,0 +1,98 @@
+module m_ordering
+
+   use m_common, only: dp, dir_X, dir_Y, dir_Z, &
+                       RDR_X2Y, RDR_X2Z, RDR_Y2X, RDR_Y2Z, RDR_Z2X, RDR_Z2Y
+
+   implicit none
+contains
+   !!
+   !! "Application storage" stores spatial data with a directionality for better cache locality
+   !!  This set of functions converts indices from this application storage (_dir) to cartesian indices (_ijk)
+   !! 
+
+   ! Get cartesian index from application storage directional one
```

Comment:
Moving this comment just below the subroutine decleration and starting with double `!!` triggers ford to create documentation automatically for us. See https://xcompact3d.github.io/x3d2/developer/style.html#in-code-docs